designs/configurable-registeration-ttl.md

```diff
@@ -0,0 +1,35 @@
+# RFC: Configurable registeration TTL
+
+## Motivation
+
+[Issue](https://github.com/kubernetes-sigs/karpenter/issues/357)
+
+Karpenter currently hardcodes [registeration TTL](https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/controllers/nodeclaim/lifecycle/liveness.go#L44) which means nodes must take < 15 minutes to come up and register with the Kubernetes control plane.
+
+This is problematic because some users have nodes that take longer that time to register. As mentioned in the issue, nodes that run GPU can often take > 15 minutes to start up. 
+
+## Configuration
+### Option 1:
+#### Introduce a configuration registeration TTL to NodeClass and propagate it to the NodeClaim
+
+We will keep a default for all nodes at 15 minute TTL however in the cases that the NodeClass would be used with GPUs or known to take a longer time to register, we will expose a new configuration registerationTTL. We will allow users to configure their own TTL without presets
+
+```yaml
+spec:
+  registerationTTL:  30m
+```
+
+#### Evaluation conditions 
+
+1. When evaluating liveness, check if NodeClaim contains a registerationTTL if so utilize that TTL instead of the default
+
+### Option 2 (Preferred):
+#### Introduce a configuration registeration TTL to Node Pool then propagate to node claim
+
+We would introduce the new argument in Node Pool and perform similar propagation of the argument to Node Claim. In addition, we won't consider this variable when evaluating drift as 
+this variable is only used duration registration. If a node needed a longer TTL then the node would have been killed by Karpenter and if it did register then the TTL didn't matter, so a change to this variable wouldn't require Drift to update.
```

Comment:
Discussed offline to add a few more details on the stance we are going to take updates for unregisterd nodeclaim that are out of sync with the nodepool.